SonarQube.Client.Tests/Models/ServerSentEvents/TaintVulnerabilityRaisedServerEventTests.cs

```diff
@@ -31,25 +33,126 @@ public class TaintVulnerabilityRaisedServerEventTests
         [TestMethod]
         public void Ctor_InvalidProjectKey_Throws()
         {
-            Action act = () => { new TaintVulnerabilityRaisedServerEvent(null, "id1", "main"); };
+            Action act = () => { new TaintVulnerabilityRaisedServerEvent(null, "id1", "main", "ruleKey", SonarQubeIssueSeverity.Blocker, SonarQubeIssueType.Vulnerability, null, null); };
 
             act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("projectKey");
         }
 
         [TestMethod]
         public void Ctor_InvalidKey_Throws()
         {
-            Action act = () => { new TaintVulnerabilityRaisedServerEvent("MyProject", null, "main"); };
+            Action act = () => { new TaintVulnerabilityRaisedServerEvent("MyProject", null, "main", "ruleKey", SonarQubeIssueSeverity.Blocker, SonarQubeIssueType.Vulnerability, null, null); };
 
             act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("key");
         }
 
         [TestMethod]
         public void Ctor_InvalidBranch_Throws()
         {
-            Action act = () => { new TaintVulnerabilityRaisedServerEvent("MyProject", "id1", null); };
+            Action act = () => { new TaintVulnerabilityRaisedServerEvent("MyProject", "id1", null, "ruleKey", SonarQubeIssueSeverity.Blocker, SonarQubeIssueType.Vulnerability, null, null); };
 
             act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("branch");
         }
+
+        [TestMethod]
+        public void Deserialize_ServerEventDeserializedCorrectly()
+        {
+            const string serializedTaintVulnerabilityRaisedEvent = @"{
+	""key"": ""taintKey"",
+	""projectKey"": ""projectKey1"",
+	""branch"": ""master"",
+	""creationDate"": 123456789,
+	""ruleKey"": ""javasecurity:S123"",
+	""severity"": ""MAJOR"",
+	""type"": ""VULNERABILITY"",
+	""mainLocation"": {
+		""filePath"": ""functions/taint.js"",
+		""message"": ""blah blah"",
+		""textRange"": {
+			""startLine"": 17,
+			""startLineOffset"": 10,
+			""endLine"": 3,
+			""endLineOffset"": 2,
+			""hash"": ""hash""
+		}
+	},
+	""flows"": [
+		{
+			""locations"": [
+				{
+					""filePath"": ""functions/taint.js"",
+					""message"": ""sink: tainted value is used to perform a security-sensitive operation"",
+					""textRange"": {
+						""startLine"": 15,
+						""startLineOffset"": 14,
+						""endLine"": 6,
+						""endLineOffset"": 7,
+						""hash"": ""hash1""
+					}
+				},
+				{
+					""filePath"": ""functions/taint2.js"",
+					""message"": ""sink: tainted value is used to perform a security-sensitive operation"",
+					""textRange"": {
+						""startLine"": 18,
+						""startLineOffset"": 11,
+						""endLine"": 4,
+						""endLineOffset"": 3,
+						""hash"": ""hash2""
+					}
+				}
+			]
+		}
+	]
+}";
+
+            var result = JsonConvert.DeserializeObject<TaintVulnerabilityRaisedServerEvent>(serializedTaintVulnerabilityRaisedEvent);
```

Comment:
The assertions are disgusting but I don't have a better solution

Reply:
Maybe like this?
```
var deserialized = JsonConvert.DeserializeObject<T>(JsonConvert.SerializeObject<T>(original));
deserialized.Should().BeEquivalentTo(original);
```